### PR TITLE
Reorder and revamp the documentation

### PR DESCRIPTION
**Pull request by glonlas**
This PR improve the documentation by trying to explain in details all steps to set up the bot.

## Broken element in this branch
You will see on README.md the image link and link to documentation pages are broken. This is normal. If you look closely the URL links to `/develop/` branch.

## Missings parts
- How to use hyperopt and how to read the results. It will be part of another PR
- Template for sending an issue and a pull-request

### CODE REVIEW

**Comment by vertti** [comment]
Hyperopt using is described in our wiki. https://github.com/gcarq/freqtrade/wiki/Testing-new-strategies-with-Hyperopt

**Comment by glonlas** [comment]
I have to integrate it into this PR. One goal of this PR is to allow users to improve the documentation and also be able to track modifications.

File: README.md [hunk 1]
```diff
@@ -1,160 +1,132 @@
...
+Simple High frequency trading bot for crypto currencies designed to 
+support multi exchanges and be controlled via Telegram.
+
+![freqtrade](https://raw.githubusercontent.com/gcarq/freqtrade/develop/docs/assets/freqtrade-screenshot.png)
+
+## Disclaimer
+This software is for educational purposes only. Do not risk money which 
+you are afraid to lose. Always start by running a trading bot in Dry-run and do not engage money
+before you understand how it works and what profit/loss you should
+expect.
```
Comment (vertti) [inline README.md]: I wonder if we should use this opportunity to add a final note along the lines of USE THE SOFTWARE AT YOUR OWN RISK. THE AUTHORS AND ALL AFFILIATES ASSUME NO RESPONSIBILITY FOR YOUR TRADING RESULTS.

Comment (vertti) [reply README.md]: I really like your additions here though.

File: docs/bot-optimization.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
+# Bot Optimization
```
Comment (vertti) [inline docs/bot-optimization.md]: I think these should be Github wiki pages that we link to from README.md.

**Comment by shusso** [comment]
Good stuff @glonlas 👍

**Comment by vertti** [comment]
But damn, all in all that was one awesome set of new documentation, good work!

File: README.md [hunk 3]
```diff
@@ -1,160 +1,132 @@
...
+
+## Branches
+The project is currently setup in two main branches:
+- `develop` - This branch has often new features, but might also cause 
+breaking changes.
+- `master` - This branch contains the latest stable release. The bot 
+'should' be stable on this branch, and is generally well tested. 
+
+## Features
+- [x] **Based on Python**: For botting on any operating system -
```
Comment (shusso) [inline README.md]: IIRC we require python3.6+ which should be mentioned somewhere

File: docs/bot-optimization.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+```
+
+**Reload your testdata files**
+```bash
+python3 ./freqtrade/main.py -c config.json backtesting --realistic-simulation --refresh-pairs-cached
+```
+
+**With live data (do not alter your testdata files)**
+```bash
+python3 ./freqtrade/main.py -c config.json backtesting --realistic-simulation --live
```
Comment (shusso) [inline docs/bot-optimization.md]: do we need to pass the `-c` here, shouldn't we use the config by default in backtestging/hyperopt?

File: docs/installation.md [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+3. Go into the new directory
+```bash
+cd freqtrade
+```
+4. Copy `config.sample` to `config.json`
+```bash
+cp config.json.example config.json
+```
+To edit the config please refer to the [Bot Configuration](https://github.com/gcarq/freqtrade/blob/develop/docs/configuration.md) page 
+5. Create your DB file
```
Comment (shusso) [inline docs/installation.md]: aren't these created automatically?

File: docs/installation.md [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+
+### 2.3. MacOS installation
+
+**2.3.1. Install git and wget**
+```bash
+brew install wget git
+```
+
+**2.3.2. Install Python 3.6**  
+- [Python 3.6 MacOS offical page](https://www.python.org/downloads/mac-osx/)
```
Comment (shusso) [inline docs/installation.md]: IIRC on osx you can just `brew install python3` to get the latest released version of python

File: docs/installation.md [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+```
+To edit the config please refer to [Bot Configuration](https://github.com/gcarq/freqtrade/blob/develop/docs/configuration.md) 
+
+## 5. Setup your virtual env
+```bash
+python3.6 -m venv .env
+source .env/bin/activate
+pip3.6 install -r requirements.txt
+pip3.6 install -e .
+pip3.6 install hyperopt
```
Comment (shusso) [inline docs/installation.md]: Not needed, as hyperopt is in `requirements.txt`

File: docs/installation.md (hunk 6 above)
Comment (vertti) [reply docs/installation.md]: Yep, that is the preferable method.

File: docs/bot-optimization.md (hunk 2 above)
Comment (glonlas) [reply docs/bot-optimization.md]: The issue with the wiki is that no-one can edit it if you are not an admin. And even for us we cannot review and roll it back.

I would like to give the chance from non-tech to contribute to the doc.

**Comment by glonlas** [comment]
Thank you very much for your review and comments. I am updating the PR, then we can merge it.

File: README.md (hunk 1 above)
Comment (glonlas) [reply README.md]: I like it. Added to the documentation.

File: README.md (hunk 3 above)
Comment (glonlas) [reply README.md]: I have added a whole section about the software requirements in the README.md

File: docs/bot-optimization.md (hunk 4 above)
Comment (glonlas) [reply docs/bot-optimization.md]: Removed

File: docs/installation.md (hunk 5 above)
Comment (glonlas) [reply docs/installation.md]: Fixed

File: docs/installation.md (hunk 6 above)
Comment (glonlas) [reply docs/installation.md]: Fixed

File: docs/installation.md (hunk 7 above)
Comment (glonlas) [reply docs/installation.md]: Removed

**Pull request merged**